### ISSUE TITLE
the request body couldn't be sent correctly if content set to 'application/json; charset=utf-8'

### ISSUE THREAD

**Issue by lopaz** [elbywan/wretch#90]
the following code doesn't work:
`wretch().content('application/json; charset=utf-8').url('http://...').post(json)`

it seems that `charset=utf-8` is not allowed:  

https://github.com/elbywan/wretch/blob/2631890ddcd0450b66d6a0eb9670012c73354b2e/src/wretcher.ts#L199-L206
https://github.com/elbywan/wretch/blob/2631890ddcd0450b66d6a0eb9670012c73354b2e/src/wretcher.ts#L12-L13

not sure if this is a bug, since json is the default content-type and utf-8 is the default chareset for json, which means `content('application/json; charset=utf-8')` is totally redundant.

**Comment by elbywan** [comment]
Hi @lopaz, thanks for reporting the issue!

> not sure if this is a bug, since json is the default content-type and utf-8 is the default chareset for json, which means content('application/json; charset=utf-8') is totally redundant.

I think this is a bug, `wretch` should never override a `Content-Type` header having a custom charset. I'll push a fix to prevent this behaviour!

With that said, in theory according to [`RFC-4627`](https://tools.ietf.org/html/rfc4627) for the `application/json` content type the encoding is determined by the first four bytes of the payload:

```
JSON text SHALL be encoded in Unicode.  The default encoding is UTF-8.

Since the first two characters of a JSON text will always be ASCII
characters [RFC0020], it is possible to determine whether an octet
stream is UTF-8, UTF-16 (BE or LE), or UTF-32 (BE or LE) by looking
at the pattern of nulls in the first four octets.
```

I also noticed that browsers often override the content-type value regardless of the fetch options. So in practice, in that context omitting the charset should not matter since browsers add the value themselves.

**Comment by elbywan** [comment]
I just released version `1.7.3` with the fix 📦.

**Comment by lopaz** [comment]
@elbywan  Good job. I didn't realize the charset-overwritten issue. 

There's still another one, using `wretch().content('application/json; charset=utf-8').url('http://...').post({foo: 'bar'})`, the request payload would be `[object Object]` instead of `{foo: 'bar'}`, which made the request failed.
![image](https://user-images.githubusercontent.com/8480947/95960570-780be380-0e36-11eb-8b05-77e7d01f393f.png)

**Comment by elbywan** [comment]
@lopaz Good catch! I pushed a fix that should solve the problem.

**Comment by lopaz** [comment]
Confirmed, fixed in 1.7.4  👍
